### PR TITLE
Fix lottery contract

### PR DESCRIPTION
**Pull request by AlexFox86**
Fix Action and Event enums.
Remove get_balance(), get_players().
Fix tests.

### CODE REVIEW

File: lottery/Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ gstd = { git = "https://github.com/gear-tech/gear.git", features = ["debug"] }
 codec = { package = "parity-scale-codec", version = "3.1.2", default-features = false, features = ["derive", "full"] }
 scale-info = { version = "2.0.1", default-features = false, features = ["derive"] }
 primitive-types = { version = "0.11.1", default-features = false, features = ["scale-info"] }
-sp-core-hashing = { version = "4.0.0", default-features = false }
+sp-core = { version = "6.0.0", git = "https://github.com/gear-tech/substrate.git", branch = "gear-stable", default-features = false, features = ["full_crypto"] }
```
Comment (shamilsan) [inline lottery/Cargo.toml:13]: Why do you use heavier `sp-core` instead of lighter `sp-core-hashing`?

File: lottery/src/lib.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ use ft_io::*;
 use gstd::{debug, exec, msg, prelude::*, ActorId};
 use lt_io::*;
 use scale_info::TypeInfo;
+use sp_core::hashing::blake2_256;
```
Comment (shamilsan) [inline lottery/src/lib.rs:17]: Please follow the idiomatic `use`.

https://doc.rust-lang.org/stable/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html#creating-idiomatic-use-paths

```suggestion
use sp_core::hashing;
```

File: lottery/src/lib.rs [hunk 3]
```diff
@@ -122,80 +123,10 @@ impl Lottery {
...
-                self.players.is_empty()
-            );
-        }
-    }
-
     // Random number generation from block timestamp
     fn get_random_number(&mut self) -> u32 {
         let timestamp: u64 = exec::block_timestamp();
-        let code_hash = sp_core_hashing::blake2_256(&timestamp.to_be_bytes());
+        let code_hash = blake2_256(&timestamp.to_be_bytes());
```
Comment (shamilsan) [inline lottery/src/lib.rs:129]: Idiomatic call of extern functions is:
```suggestion
        let code_hash = hashing::blake2_256(&timestamp.to_be_bytes());
```

File: lottery/src/lib.rs [hunk 4]
```diff
@@ -122,80 +123,10 @@ impl Lottery {
...
     }
 
-    /// Removes player from lottery
-    /// Requirements:
-    /// * Lottery has started and lottery time has not expired
-    /// * Player must be on the player list
-    /// * Player can only remove himself
-    /// Arguments:
-    /// * `index`: lottery player index
-    async fn leave_lottery(&mut self, index: u32) {
```
[the remark this answers is not in the document]
Comment (shamilsan) [reply lottery/src/lib.rs:132]: Discussed, this is a change in logic.

File: lottery/Cargo.toml (hunk 1 above)
Comment (AlexFox86) [reply lottery/Cargo.toml]: Fixed

File: lottery/src/lib.rs (hunk 2 above)
Comment (AlexFox86) [reply lottery/src/lib.rs]: Fixed

File: lottery/src/lib.rs (hunk 3 above)
Comment (AlexFox86) [reply lottery/src/lib.rs]: Fixed

**Review by shamilsan — APPROVED** [review]

**Review by LouiseMedova — APPROVED** [review]

**Pull request merged**